enumerables.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  # rubocop:disable Style/For
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in 0...size
+      yield(to_a[i])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    for i in 0...size
+      yield(to_a[i], i)
+    end
+    nil
+  end
+
+  # rubocop:enable Style/For
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    res = []
+    my_each do |x|
+      res.append(x) if yield(x)
+    end
+    res
+  end
+
+  # rubocop: disable Style/CaseEquality, Style/IfInsideElse
+  def my_all?(args = nil)
+    res = true
+    my_each do |x|
+      if block_given?
+        res = false unless yield(x)
+      elsif !args
+        res = false unless x
+      else
+        res = false unless args === x
+      end
+    end
+    res
+  end
+
+  def my_any?(args = nil)
+    res = false
+    my_each do |x|
+      if block_given?
+        res = true if yield(x)
+      elsif !args
+        res = true if x
+      else
+        res = true if args === x
+      end
+    end
+    res
+  end
+
+  def my_none?(args = nil)
+    res = true
+    my_each do |x|
+      if block_given?
+        res = false if yield(x)
+      elsif !args
+        res = false if x
+      else
+        res = false if args === x
+      end
+    end
+    res
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
  !my_any?(arg, &block)
end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?

Reply:
that's a cool hack, thanks for pointing out :+1: